### PR TITLE
code hygiene:  propTypes in components

### PR DESCRIPTION
**Pull request by RedSquirrelious**

### CODE REVIEW

File: test/services/PlaybookServiceTest.js [hunk 1]
```diff
@@ -37,7 +37,10 @@ describe('Playbook Service', function () {
                 dateTimeComparisonValue: '19700101',
                 dataFormat: dateTimeDataFormatValue
             })
-            expect(playbook.GetComparisonValue(getComparisonValueDateTime).isSame('19700101')).to.be.true
+
```
Comment (strand) [inline test/services/PlaybookServiceTest.js]: I don't see why this revision needs to occur within this propType work. Is this related? If not, please restore to the master version.

File: test/components/Auth/EnsureLoggedInTest.js [hunk 2]
```diff
@@ -47,4 +49,78 @@ describe('EnsureLoggedIn', function test () {
     it('Should render children when no error is present and user is logged in', () => {
         expect(this.loggedInOutput.type).to.equal('div')
     })
+
```
Comment (strand) [inline test/components/Auth/EnsureLoggedInTest.js]: There's a missing context here, which I can tell because there is shared setup between two context or describe blocks. I believe its title is 'when the console has an error'

File: src/reducers/index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { combineReducers } from 'redux'
```
Comment (strand) [inline src/reducers/index.js]: I don't see why this revision needs to occur within this propType work. Is this related? If not, please restore to the master version.

File: src/configureStore.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { createStore, applyMiddleware, compose, combineReducers } from 'redux'
+import { createStore, applyMiddleware, compose } from 'redux'
```
Comment (strand) [inline src/configureStore.js]: I don't see why this revision needs to occur within this propType work. Is this related? If not, please restore to the master version.

File: src/components/Engagements.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ export const mapStateToPropsEngagements = (state, ownProps) => {
     }
 }
 
-export const Engagements = ({dispatch, incidentId, engagements, user}) => {
+export const Engagements = ({ dispatch, incidentId, engagements, user }) => {
```
Comment (strand) [inline src/components/Engagements.js]: I don't see why this revision needs to occur within this propType work. Is this related? If not, please restore to the master version.

File: src/actions/incidentActions.js [hunk 6]
```diff
@@ -151,5 +151,5 @@ export const updateIncidentCreationInput = (input) => ({
 
 export const duplicateIncident = (ticketId) => (dispatch) => {
     dispatch(createIncidentActionSet(ticketId, {}).fail({ message: 'An incident already exists for this ticket'}))
-    dispatch(ticketActions.updateIncidentQuery(ticketId))
+    dispatch(ticketActions.updateTicketQuery(ticketId))
```
Comment (strand) [inline src/actions/incidentActions.js]: I don't see why this revision needs to occur within this propType work. Is this related? If not, please restore to the master version.

File: test/components/Auth/EnsureLoggedInTest.js [hunk 7]
```diff
@@ -47,4 +49,78 @@ describe('EnsureLoggedIn', function test () {
...
+        let correctType = {
+            error: 'TESTTEST',
+            isLoggedIn: true,
+            store: GetMockStore({ auth: { error: 'Test Error' } })
+        }
+
+        describe('correct propTypes', function () {
+            it('should return false for no error with the propType', function () {
+                let result = setup(correctType, children)
+                let p = isPropTypeWarningMatch(console.error.args.toString())
```
Comment (strand) [inline test/components/Auth/EnsureLoggedInTest.js]: Never use single letter variables. `hasWarning` would be a fine name for this.

File: test/test-utils.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import sinon from 'sinon'
+
+export const isPropTypeWarningMatch = (warning) => {
+    return typeof(warning) === 'string' &&
+        warning.indexOf('Warning: Failed prop type:') > -1
+}
+
+
+export const stubConsoleError = () => {
```
Comment (strand) [inline test/test-utils.js]: Don't abstract before/after blocks from their test context, this is important info which should be within your specs.

File: test/components/Auth/EnsureLoggedInTest.js [hunk 9]
```diff
@@ -47,4 +49,78 @@ describe('EnsureLoggedIn', function test () {
...
+
+                sinon.assert.notCalled(console.error)
+                expect(p).to.be.false
+            })
+        })
+    })
+
+    context('propTypes passed in wrong', function() {
```
Comment (strand) [inline test/components/Auth/EnsureLoggedInTest.js]: As far as I can tell, these specs test the propTypes modules implementation. When testing, only test code you control. You aren't a propType maintainer, so don't test the behavior proTyprs provide.

File: src/actions/incidentActions.js (hunk 6 above)
Comment (RedSquirrelious) [reply src/actions/incidentActions.js]: There is no updateIncidentQuery in ticketActions.  Reverted back to master for now.  Can be cleaned up later.

File: src/components/Engagements.js (hunk 5 above)
Comment (RedSquirrelious) [reply src/components/Engagements.js]: No problem.  Reverted

File: src/configureStore.js (hunk 4 above)
Comment (RedSquirrelious) [reply src/configureStore.js]: combineReducers isn't used, resulting in linter errors.  Reverted to master to help clarify this PR.

File: src/reducers/index.js (hunk 3 above)
Comment (RedSquirrelious) [reply src/reducers/index.js]: It was missing and causing linting errors.  It isn't needed because the filter reducer is combined with the event reducer in the event reducer file.  Someone else already fixed it on master.  Reverted.

File: test/components/Auth/EnsureLoggedInTest.js (hunk 2 above)
Comment (RedSquirrelious) [reply test/components/Auth/EnsureLoggedInTest.js]: The consensus seems to be that we are not going to test for propTypes.  Tests removed accordingly.  :)

File: src/configureStore.js (hunk 4 above)
Comment (RedSquirrelious) [reply src/configureStore.js]: Actually, someone removed combineReducers already from master.  Not sure why this is showing us as a difference.

**Comment by RedSquirrelious** [comment]
Branch had some odd issues with master.  Strand rebased a version on their computer, which fixed the issue.  See PR # 57.
